src/west/manifest.py

```diff
@@ -228,6 +233,24 @@ def __init__(self, source_file=None, source_data=None,
             self._malformed('manifest contains no manifest element')
         data = self._data['manifest']
 
+        # Make sure this version of west can load this manifest data.
+        # This has to happen before the schema check -- later schemas
+        # are likely to incompatibly extend our own.
+        if 'version' in data:
+            min_version = data['version']
+            # As a convenience for the user, convert floats to strings.
+            # This avoids forcing them to write:
+            #
+            #  version: "1.0"
+            #
+            # by explicitly allowing:
+            #
+            #  version: 1.0
+            if isinstance(min_version, float):
+                min_version = str(min_version)
+            if parse_version(min_version) > _WEST_VERSION:
```

Comment:
What happens with pre-release versions that have additional metadata after the semantic version? i.e. `0.6.3-rc1` for example.

Reply:
They just work -- see the test cases.